src/Features/Core/Portable/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer.InProcOrRemoteHostAnalyzerRunner.cs

```diff
@@ -149,7 +149,7 @@ private async Task FireAndForgetReportAnalyzerPerformanceAsync(Project project,
                 {
                     var analyzerMap = pooledObject.Object;
 
-                    analyzerMap.AppendAnalyzerMap(analyzerDriver.Analyzers.Where(a => !a.IsOpenFileOnly(project.Solution.Workspace)));
+                    analyzerMap.AppendAnalyzerMap(analyzerDriver.Analyzers.Where(a => !a.IsInProcessOnly() && (forcedAnalysis || !a.IsOpenFileOnly(solution.Workspace))));
```

Comment:
I am not sure what "forcedAnalysis" is used here. so, analyzer which is marked as open file only will be asked to run in out of proc if forced is true? 

as I said above, in proc: false && open file only: true doesn't make sense for OOP.

Reply:
> so, analyzer which is marked as open file only will be asked to run in out of proc if forced is true?

`forcedAnalysis` forces Open File Only to false. The full-scope analysis then proceeds in-process or out-of-process as specified by the analyzer.

Reply:
okay, now I understand more on your intention.

I think here. you can remove IsOpenFileOnly call completely. and just check forcedAnalysys && InProce

and out of proc, you just check out of proc, you don't even need to check forced analysis.